docs/source/hyper-parameter-search.rst

```diff
@@ -1,33 +1,193 @@
 Hyper Parameter Search
 ======================
 
-*Tools for performing hyperparameter optimization of Scikit-Learn API-compatible models using Dask*.
+*Tools to perform hyperparameter optimizaiton of Scikit-Learn API-compatible
+models using Dask, and to scale hyperparameter optimization to either* **larger
+data** *or* **more computational power.**
 
-There are two kinds of hyperparameter optimization estimators
-in Dask-ML. The appropriate one to use depends on the size of your dataset and
-whether the underlying estimator implements the `partial_fit` method.
+Scaling hyperparameter searches
+-------------------------------
 
-If your dataset is relatively small or the underlying estimator doesn't implement
-``partial_fit``, you can use :class:`dask_ml.model_selection.GridSearchCV` or
-:class:`dask_ml.model_selection.RandomizedSearchCV`.
-These are drop-in replacements for their scikit-learn counterparts, that should offer better performance and handling of Dask Arrays and DataFrames.
-The underlying estimator will need to be able to train on each cross-validation split of the data.
-See :ref:`hyperparameter.drop-in` for more.
+Dask-ML has a ton of tools to get around common issues in hyperparameter
+optimization:
 
-If your data is large and the underlying estimator implements ``partial_fit``, you can
-Dask-ML's :ref:`*incremental* hyperparameter optimizers <hyperparameter.incremental>`.
+.. autosummary::
+   dask_ml.model_selection.GridSearchCV
+   dask_ml.model_selection.RandomizedSearchCV
+   dask_ml.model_selection.IncrementalSearchCV
+   dask_ml.model_selection.HyperbandSearchCV
+   dask_ml.model_selection.SuccessiveHalvingSearchCV
+
+These avoid the two most common issues in hyperparameter optimization, when
+the hyperparameter search is...
+
+1. "**memory constrained"**, which happens when the dataset size is too large
+   to fit in memory.  This typically happens when a model needs to be tuned for
+   a larger-than-memory dataset after local development.
+2. "**compute constrained**", which happen when the computation takes too long
+   even with data that can fit in memory.  This typically happens when many
+   hyperparameters need to be tuned, which especially common with neural
+   networks.
+
+Here's an example of what "memory constrained" means:
```

Comment:
A bit of prose on each of these would be welcome. For example

- Memory contained: "The original DataFrame or array is larger than your single machine's memory"
- Compute constrained: "The dataset fits in memory, but training takes too long."